adr/2024_05_15_use_separate_base_form_types_for_admin_panel_instead_of_type_extensions.md

```diff
@@ -0,0 +1,55 @@
+# Use Separate Base Form Types Instead of Type Extensions
+
+* Status: Accepted
+* Date: 2024-05-15
+
+## Context and Problem Statement
+
+In our current system, every base form type is placed at the lowest possible level, either within a specific bundle or core. 
+Form extensions serve as the primary means of extending form types since all form types are final. 
+However, this approach leads to a significant issue: form extensions work globally, which prevents us from reusing any form type easily in different contexts. 
+For example, if a field is added through an extension in the AdminBundle (admin context), the same field will also appear in the shop's context.
+This problem is compounded by the introduction of Symfony UX, which necessitates overriding some fields to enable its functionality. 
+As a result, distinguishing which context added a specific change and when each change should be applied becomes exceedingly difficult.
+
+## Decision Drivers
+
+* Need for context-specific form type customization.
+* Avoiding global side effects caused by form extensions.
+* Clear separation between admin and shop form types.
+* Simplification of managing and overriding form fields with Symfony UX.
+
+## Considered Options
+
+### Option 1: Use Separate Base Form Types
+
+* **Good, because:**
+    * Provides clear separation between admin and shop form types.
+    * Allows context-specific customization without affecting other contexts.
+    * Simplifies debugging and maintenance by isolating changes to their respective contexts.
+    * Facilitates the integration and management of Symfony UX-specific overrides.
+
+* **Bad, because:**
+    * Increases the number of form types, leading to potential redundancy.
+    * Requires additional effort to create and maintain separate form types.
+
+### Option 2: Continue Using Type Extensions
+
+* **Good, because:**
+    * Fewer form types to manage.
+    * Leverages existing infrastructure without requiring significant changes.
+
+* **Bad, because:**
+    * Form extensions affect all contexts globally, leading to unintended side effects.
+    * Difficult to distinguish and manage context-specific changes.
+    * Complicates integration with Symfony UX in AdminBundle due to overlapping overrides.
+
+## Decision Outcome
+
+Chosen option: "Option 1: Use Separate Base Form Types for Admin Panel", because it provides a clear and manageable way to customize form types for different contexts. 
```

Comment:
```suggestion
Chosen option: "Use Separate Form Types", because it provides a clear and manageable way to customize form types for different contexts.
```